src/main/scala/za/co/absa/standardization/types/TypedStructField.scala

```diff
@@ -271,14 +273,14 @@ object TypedStructField {
 
     private val radix: Radix = readRadixFromMetadata
 
-    override val parser: Try[IntegralParser[L]] = {
+    override def parser(origType: DataType): Try[IntegralParser[L]] = {
       pattern.flatMap { patternForParser =>
       if (radix != Radix.DefaultRadix) {
         val decimalSymbols = patternForParser.map(_.decimalSymbols).getOrElse(defaults.getDecimalSymbols)
-        Try(IntegralParser.ofRadix(radix, decimalSymbols, Option(typeMin), Option(typeMax)))
+        Try(IntegralParser.ofRadix(origType.typeName, structField.dataType.typeName, radix, decimalSymbols, Option(typeMin), Option(typeMax)))
       } else {
-        Success(IntegralParser(patternForParser
-          .getOrElse(NumericPattern(defaults.getDecimalSymbols)), Option(typeMin), Option(typeMax)))
+        Success(IntegralParser(origType.typeName, structField.dataType.typeName,
+          patternForParser.getOrElse(NumericPattern(defaults.getDecimalSymbols)), Option(typeMin), Option(typeMax)))
```

Comment:
For better readability I would suggest:

```suggestion
        Success(IntegralParser(
          origType.typeName, 
          structField.dataType.typeName,
          patternForParser.getOrElse(NumericPattern(defaults.getDecimalSymbols)), 
          Option(typeMin), 
          Option(typeMax)
        ))
```